common/djangoapps/xblock_django/user_service.py

```diff
@@ -35,7 +35,7 @@ def get_current_user(self):
         """
         return self._convert_django_user_to_xblock_user(self._django_user)
 
-    def get_anonymous_user_id(self, username, course_id):
+    def get_anonymous_user_id(self, username, course_id, ignore_staff_check=False):
```

Comment:
I was trying to figure this out this morning, but got side-tracked by other work.  If there's a way to introspect the XBlock that is currently using this service, I think it'd be helpful to do that here and check against a list of xblock types that are allowed to call this function with `ignore_staff_check=True`.  There's a clue in here about looking back from a service to the calling xblock, but not sure if it's fully-baked: https://github.com/edx/XBlock/blob/master/xblock/reference/plugins.py You should be able to play around in a python debugger and figure it out, I think.

This would at least make it a little more clear _which_ thing is circumventing the privilege-checking below.